### PR TITLE
IPQ 02: Linked List Intersection

### PR DESCRIPTION
**Pull request by char-adadev**
The purpose of this PR is to introduce the Linked List Intersection IPQ.

For review:
* Please run learn preview
* Verify the example implementation passes tests
* Any suggestions for wording, grammar, etc are welcome

### CODE REVIEW

File: 02/intersection.md [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+For example, while the following linked lists have tails that share the same values, they are not considered to intersect because the nodes of the linked lists are not the same nodes in memory.
+
+![intersecting linked lists example 3](../images/intersection_linked_list_example_3.png)
+
+There are no cycles anywhere in the linked list structures. Assume any intersection includes the tails of each list.
+
+The linked lists must retain their original structure after the function returns.
+
+<br>
+<details style="max-width: 700px; margin: auto;">
```
Comment (kyra-at-ada) [inline 02/intersection.md:26]: Is it possible to move this inside the checkpoint? I feel like we don't want to be providing them with a comprehensive set of test cases right before we ask them to write their own?

Comment (kyra-at-ada) [reply 02/intersection.md:26]: Alternatively, we could ask them to write test cases different than the ones we provide, but I feel like that might get tricky with the edge cases

File: 02/intersection.md [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+
+##### !end-question
+
+##### !placeholder
+
+Write the logical steps here.
+
+##### !end-placeholder
+
+### !end-challenge
```
Comment (kyra-at-ada) [inline 02/intersection.md:347]: Did you make a decision not to add sample logical steps in the explanation like we do in core PSEs? I don't remember if we actually discussed this.

File: 02/implement-intersection.checkpoint.md [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+        # Assert
+        self.assertEqual(answer, None)
+```
+### !end-tests
+### !explanation
+
+An example of a working implementation:
+
+```python
+def intersection_node(head_a, head_b):
```
Comment (kyra-at-ada) [inline 02/implement-intersection.checkpoint.md:217]: Consider commenting this to deepen student understanding. I think this could be useful if we don't get around to making video walkthroughs of possible solutions

Comment (kyra-at-ada) [reply 02/implement-intersection.checkpoint.md:217]: Also consider adding explanation of time/space complexity of this solution? Is that too much of a giveaway?

**Review by kyra-at-ada — APPROVED** [review]
💫 This looks great Char! Added some questions/suggestions, but even without changes looks good to merge. 

Checked the instructor solution against the checkpoint.

File: 02/intersection.md (hunk 2 above)
Comment (char-adadev) [reply 02/intersection.md:347]: Yes, I decided not to include the sample logical steps because I saw them as hints and wanted to challenge the interns to come up with logical steps from scratch. That being said, I can add them here (and to the Pascal's triangle IPQ as well) and folx can choose to access them whenever needed 🙌🏾

File: 02/implement-intersection.checkpoint.md (hunk 3 above)
Comment (char-adadev) [reply 02/implement-intersection.checkpoint.md:219]: Added comments and time/space complexity!
